app/javascript/src/utils/debounceAction.js

```diff
@@ -0,0 +1,24 @@
+import debounce from 'lodash.debounce';
+
+/**
+ * A util function to wrap any action with `debounce`.
+ * To use this util, wrap it in the util like so: debounceAction(onSearchBoxType.bind(this));
+ *
+ * By default, this util uses a default time of 300ms, and includes a config of `{ leading: true }`,
+ * which will pass the first received value to debounced function. These values can be overridden:
+ * debounceAction(this.onSearchBoxType.bind(this), 150);
+ *
+ *
+ * @param {Function} action - The function that should be wrapped with `debounce`.
+ * @param {Number} [debounceTime=300] - The number of milliseconds to wait.
+ * @param {Object} [debounceConfig={ leading: true }] - Any configuration for the debounce function.
+ *
+ * @returns {Function} A function wrapped in `debounce`.
+ */
+export default function debounceAction(
+  action,
+  debounceTime = 300,
+  debounceConfig = { leading: true },
```

Comment:
For this config, is there any way that we could take the config passed in and merge it with the default? I could see someone setting a different option besides leading and then having that wipe out this leading default.

Reply:
@mstruve Ah, that's a good call. Makes sense for the `debounceConfig` especially (was just thinking of the `debounceTime` when I wrote this I guess !). I'll update so that the configs can be merged ✨

Reply:
I think leading should default to false though 😊